source/agora/consensus/protocol/Nominator.d

```diff
@@ -1042,16 +1036,17 @@ extern(D):
                 return;
             }
 
-            Block block = this.ledger.buildBlock(
+            this.pending_block = this.ledger.buildBlock(
                 externalized_tx_set, data.enrolls, data.missing_validators);
 
             // Now we add our signature and gossip to other nodes
             log.trace("ADD BLOCK SIG at height {} for this node {}", height, this.kp.address);
             const self = this.enroll_man.getEnrollmentKey();
-            this.slot_sigs[height][self] = this.signBlock(block);
-            this.ledger.addHeightAsExternalizing(height);
-            this.updateMultiSignature(block.header);
-            this.verifyBlock(block);
+            this.slot_sigs[height][self] = this.signBlock(this.pending_block);
+            this.updateMultiSignature(this.pending_block.header);
+            // Now add any early signatures we received
+            early_sigs[].each!(sig => this.receiveBlockSignature(sig));
```

Comment:
We wont gossip the early sigs this way tho, right?

Reply:
Yes it will get gossiped in `updateMultisignature` when it is handles again in `receiveBlockSignature`.